script/set-react-version.js

```diff
@@ -0,0 +1,88 @@
+'use strict'
+
+const fs = require('node:fs/promises')
+const os = require('node:os')
+const path = require('node:path')
+
+const versions = new Map([
+  [
+    '17',
+    {
+      devDependencies: [
+        {
+          name: 'react',
+          version: '17.0.2',
+        },
+        {
+          name: 'react-dom',
+          version: '17.0.2',
+        },
+        {
+          name: 'react-test-renderer',
+          version: '17.0.2',
+        },
+        {
+          name: '@testing-library/react',
+          version: '12.1.5',
+        },
+      ],
+    },
+  ],
+  [
+    '18',
+    {
+      devDependencies: [
```

Comment:
Curios why we specifically chose these dev dependencies to list here? 🤔 Are they the only ones that are effected by the React version upgrade?

Reply:
@broccolinisoup exactly, I think any other deps that we want to include should be ones that mirror support for React 17/18 otherwise we can't really ship a version of PRC that works with both versions